File: Medium/287-find-the-duplicate-number/try-3.cpp
/*
 * Problem #287: Find the Duplicate Number
 * Difficulty: Medium
 * Submission: Try 3
 * status: Accepted
 * Language: cpp
 * Date: 18/01/2025, 00:20:07
 * Link: https://leetcode.com/problems/find-the-duplicate-number/
 */

class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        unordered_map<int,int> map;
        int i;
        for(i = 0 ; i<nums.size()-1;i++){
            map[nums[i]]++;
            if(map[nums[i]]>1){
                break;
            }
        }
        return nums[i];
    }
};

File: Medium/2593-find-score-of-an-array-after-marking-all-elements/try-3.py
/*
 * Problem #2593: Find Score of an Array After Marking All Elements
 * Difficulty: Medium
 * Submission: Try 3
 * status: Accepted
 * Language: python3
 * Date: 13/12/2024, 07:44:24
 * Link: https://leetcode.com/problems/find-score-of-an-array-after-marking-all-elements/
 */

class Solution:
    def findScore(self, nums: List[int]) -> int:
        heap=sorted([(val,ind) for ind,val in enumerate(nums)])
        visit=set()
        s=0
        n=len(nums)
        for i in range(n):
            if heap[i][1] in visit :
                continue
            s=s+heap[i][0]
            visit.add(heap[i][1])
            if heap[i][1]>0 :
                visit.add(heap[i][1]-1)
            if heap[i][1]<n-1 :
                visit.add(heap[i][1]+1)
        return s
        

        



        




        
